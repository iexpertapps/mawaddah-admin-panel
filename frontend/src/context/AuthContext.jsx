import React, { createContext, useState, useContext, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import api from '../services/api';

export const AuthContext = createContext(null);

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [token, setToken] = useState(null);
  const [loading, setLoading] = useState(true);
  const navigate = useNavigate();

  // On initial load, check for a token in storage
  useEffect(() => {
    const initAuth = () => {
      const storedToken = localStorage.getItem('authToken') || sessionStorage.getItem('authToken');
      if (storedToken) {
        const storedUser = localStorage.getItem('user') || sessionStorage.getItem('user');
        setToken(storedToken);
        api.defaults.headers.common['Authorization'] = `Token ${storedToken}`;
        if (storedUser) {
          try {
            setUser(JSON.parse(storedUser));
          } catch (e) {
            setUser(null);
          }
        }
      }
      setLoading(false);
    };
    initAuth();
  }, []);

  const login = async (email, password, rememberMe) => {
    try {
      const response = await api.post('/api/auth/login/', { email, password });
      const { token: apiToken, user: apiUser } = response.data;

      setUser(apiUser);
      setToken(apiToken);
      api.defaults.headers.common['Authorization'] = `Token ${apiToken}`;

      const userString = JSON.stringify(apiUser);
      if (rememberMe) {
        localStorage.setItem('authToken', apiToken);
        localStorage.setItem('user', userString);
      } else {
        sessionStorage.setItem('authToken', apiToken);
        sessionStorage.setItem('user', userString);
      }
      
      navigate('/admin', { replace: true });
      return { success: true };
    } catch (error) {
      let errorMessage = "An unexpected error occurred.";
      if (error.response?.data) {
        const errorData = error.response.data;
        errorMessage = errorData.error || (typeof errorData === 'string' ? errorData : JSON.stringify(errorData));
      }
      return { success: false, error: errorMessage };
    }
  };

  const logout = async () => {
    setUser(null);
    setToken(null);
    try {
      await api.post('/api/auth/logout/');
    } catch (error) {
      console.error('Logout failed, but proceeding with client-side cleanup.', error);
    } finally {
      localStorage.removeItem('authToken');
      localStorage.removeItem('user');
      sessionStorage.removeItem('authToken');
      sessionStorage.removeItem('user');
      delete api.defaults.headers.common['Authorization'];
      navigate('/login', { replace: true });
    }
  };

  const value = {
    user,
    token,
    isAuthenticated: !!token,
    loading,
    login,
    logout,
  };

  return (
    <AuthContext.Provider value={value}>
      {!loading && children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  const context = useContext(AuthContext);
  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
}; 