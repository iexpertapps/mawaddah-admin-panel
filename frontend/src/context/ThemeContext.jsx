import React, { createContext, useContext, useEffect, useState } from 'react'

const ThemeContext = createContext()

export const useTheme = () => {
  const context = useContext(ThemeContext)
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider')
  }
  return context
}

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState(() => {
    // Check localStorage first, then system preference, default to light
    const savedTheme = localStorage.getItem('mawaddah-theme')
    if (savedTheme) {
      return savedTheme
    }
    
    // Check system preference
    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
      return 'dark'
    }
    
    return 'light'
  })

  const toggleTheme = () => {
    setTheme(prevTheme => {
      const newTheme = prevTheme === 'light' ? 'dark' : 'light'
      localStorage.setItem('mawaddah-theme', newTheme)
      return newTheme
    })
  }

  // Apply theme to document
  useEffect(() => {
    const root = window.document.documentElement
    root.classList.remove('light', 'dark')
    root.classList.add(theme)
  }, [theme])

  // Keyboard shortcut (Ctrl+J or Cmd+J)
  useEffect(() => {
    const handleKeyDown = (event) => {
      if ((event.ctrlKey || event.metaKey) && event.key === 'j') {
        event.preventDefault()
        toggleTheme()
      }
    }

    document.addEventListener('keydown', handleKeyDown)
    return () => document.removeEventListener('keydown', handleKeyDown)
  }, [])

  const value = {
    theme,
    toggleTheme,
    isDark: theme === 'dark'
  }

  return (
    <ThemeContext.Provider value={value}>
      {children}
    </ThemeContext.Provider>
  )
} 