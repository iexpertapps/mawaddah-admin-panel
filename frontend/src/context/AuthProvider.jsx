import React, { useState, useEffect } from 'react';
import api from '../services/api';
import AuthContext from './AuthContext';

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [token, setToken] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const initAuth = () => {
      const storedToken = localStorage.getItem('authToken') || sessionStorage.getItem('authToken');
      if (storedToken) {
        const storedUser = localStorage.getItem('user') || sessionStorage.getItem('user');
        setToken(storedToken);
        api.defaults.headers.common['Authorization'] = `Token ${storedToken}`;
        if (storedUser) {
          try {
            setUser(JSON.parse(storedUser));
          } catch (e) {
            setUser(null);
          }
        }
      }
      setLoading(false);
    };
    initAuth();
  }, []);

  const login = async (email, password, rememberMe) => {
    try {
      const response = await api.post('/api/auth/login/', { email, password });
      const { token: apiToken, user: apiUser } = response.data;

      setUser(apiUser);
      setToken(apiToken);
      api.defaults.headers.common['Authorization'] = `Token ${apiToken}`;

      const userString = JSON.stringify(apiUser);
      if (rememberMe) {
        localStorage.setItem('authToken', apiToken);
        localStorage.setItem('user', userString);
      } else {
        sessionStorage.setItem('authToken', apiToken);
        sessionStorage.setItem('user', userString);
      }
      
      // Navigation removed; let the caller handle it
      return { success: true };
    } catch (error) {
      let errorMessage = "An unexpected error occurred.";
      if (error.response?.data) {
        const errorData = error.response.data;
        errorMessage = errorData.error || (typeof errorData === 'string' ? errorData : JSON.stringify(errorData));
      }
      return { success: false, error: errorMessage };
    }
  };

  const logout = async () => {
    setUser(null);
    setToken(null);
    try {
      await api.post('/api/auth/logout/');
    } catch (error) {
      console.error('Logout failed, but proceeding with client-side cleanup.', error);
    } finally {
      localStorage.removeItem('authToken');
      localStorage.removeItem('user');
      sessionStorage.removeItem('authToken');
      sessionStorage.removeItem('user');
      delete api.defaults.headers.common['Authorization'];
      // Navigation removed; let the caller handle it
    }
  };

  const value = {
    user,
    token,
    isAuthenticated: !!token,
    loading,
    login,
    logout,
  };

  return (
    <AuthContext.Provider value={value}>
      {!loading && children}
    </AuthContext.Provider>
  );
};

export default AuthProvider; 