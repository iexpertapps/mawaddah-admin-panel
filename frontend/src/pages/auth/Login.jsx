import React, { useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { Button } from '../../components/atoms/Button';
import MawaddahInput from '../../components/atoms/mawaddah/MawaddahInput';
import { Heading, Text } from '../../components/atoms/typography';
import { useTheme } from '../../context/ThemeContext';
import useAuth from '../../context/useAuth'; // Import the unified auth hook

const Login = () => {
  const { theme, isDark } = useTheme();
  const { login } = useAuth(); // Use the login function from the hook
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    email: '',
    password: '',
    rememberMe: false,
  });
  const [isLoading, setIsLoading] = useState(false);
  const [errors, setErrors] = useState({});
  const [formError, setFormError] = useState('');

  const handleInputChange = (field, value) => {
    setFormData((prev) => ({
      ...prev,
      [field]: value,
    }));
    setErrors((prev) => ({ ...prev, [field]: undefined }));
    setFormError('');
  };

  const validate = () => {
    const newErrors = {};
    if (!formData.email) newErrors.email = 'Email is required.';
    if (!formData.password) newErrors.password = 'Password is required.';
    return newErrors;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const validationErrors = validate();
    if (Object.keys(validationErrors).length > 0) {
      setErrors(validationErrors);
      return;
    }
    setIsLoading(true);
    setFormError('');

    // Call the login function from our unified useAuth hook
    const result = await login(formData.email, formData.password, formData.rememberMe);

    if (!result.success) {
      setFormError(result.error);
      setErrors({ email: true, password: true });
    } else {
      navigate('/admin', { replace: true });
    }
    setIsLoading(false);
  };

  return (
    <div className="min-h-screen flex items-center justify-center bg-gradient-to-br from-green-50 via-white to-accent-50 dark:from-gray-900 dark:via-gray-800 dark:to-gray-900 relative overflow-hidden">
      {/* Maroon Radial Blur Accent */}
      <div className="absolute bottom-0 left-0 w-48 h-48 bg-maroon/20 blur-3xl rounded-full z-0 animate-fade-in" />

      {/* Islamic Geometric Pattern Background */}
      <div className="absolute inset-0 opacity-5 dark:opacity-10 z-0">
        <div className="absolute top-0 left-0 w-96 h-96 bg-primary rounded-full mix-blend-multiply filter blur-xl animate-pulse"></div>
        <div className="absolute top-0 right-0 w-96 h-96 bg-accent rounded-full mix-blend-multiply filter blur-xl animate-pulse animation-delay-2000"></div>
        <div className="absolute -bottom-8 left-20 w-72 h-72 bg-maroon rounded-full mix-blend-multiply filter blur-xl animate-pulse animation-delay-4000"></div>
      </div>

      {/* Decorative Islamic Pattern */}
      <div className="absolute top-8 right-8 opacity-20 dark:opacity-30 animate-islamic-rotate z-0">
        <svg width="120" height="120" viewBox="0 0 120 120" className="text-primary">
          <defs>
            <pattern id="islamic-pattern" x="0" y="0" width="40" height="40" patternUnits="userSpaceOnUse">
              <path d="M20 0L24 16L40 20L24 24L20 40L16 24L0 20L16 16Z" fill="currentColor" opacity="0.3"/>
            </pattern>
          </defs>
          <rect width="120" height="120" fill="url(#islamic-pattern)"/>
        </svg>
      </div>

      {/* Main Login Card */}
      <div className="relative z-10 w-full max-w-[400px] mx-auto px-6 py-8">
        <div className="bg-gray-100/40 dark:bg-gray-800/80 backdrop-blur-sm rounded-2xl shadow-2xl border border-white/20 dark:border-gray-700/20 px-6 py-8 space-y-8">
          {/* Logo and Brand */}
          <div className="text-center space-y-4">
            <div className="flex justify-center">
              <div className="relative animate-float transition-all duration-500 hover:scale-105 shadow-[0_0_12px_rgba(26,127,85,0.6)]">
                <img
                  src="/ic_mawaddah_180x180.png"
                  alt="Mawaddah Logo"
                  className="h-16 w-16 rounded-xl object-contain mx-auto select-none"
                  draggable="false"
                />
                {/* Decorative ring */}
                <div className="absolute -inset-1 bg-gradient-to-r from-accent to-primary rounded-2xl opacity-20 blur-sm pointer-events-none"></div>
              </div>
            </div>
            <div className="space-y-2">
              <Heading size="2xl" as="h1" className="text-center bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent">
                Mawaddah
              </Heading>
              <Text size="sm" muted className="text-center">
                Admin Dashboard
              </Text>
            </div>
          </div>

          {/* Welcome Message */}
          <div className="text-center space-y-2">
            <Heading size="lg" as="h2" className="text-gray-900 dark:text-gray-100">
              Welcome Back
            </Heading>
            <Text muted className="text-sm">
              Sign in to your account to continue
            </Text>
          </div>

          {/* Login Form */}
          {formError && (
            <div className="text-center text-red-500 text-sm mb-2">{formError}</div>
          )}
          <form onSubmit={handleSubmit} className="space-y-6">
            <div className="space-y-4">
              <MawaddahInput
                id="email"
                type="email"
                label="Email Address"
                placeholder="Enter your email"
                value={formData.email}
                onChange={(e) => handleInputChange('email', e.target.value)}
                error={errors.email}
                required
                autoComplete="username"
              />
              <MawaddahInput
                id="password"
                type="password"
                label="Password"
                placeholder="Enter your password"
                value={formData.password}
                onChange={(e) => handleInputChange('password', e.target.value)}
                error={errors.password}
                required
                autoComplete="current-password"
              />
            </div>
            {/* Remember Me and Forgot Password */}
            <div className="flex items-center justify-between">
              <label className="flex items-center space-x-2 cursor-pointer">
                <input
                  type="checkbox"
                  checked={formData.rememberMe}
                  onChange={(e) => handleInputChange('rememberMe', e.target.checked)}
                  className="w-4 h-4 text-primary border-gray-300 rounded focus:ring-primary focus:ring-2"
                />
                <Text size="sm" className="text-gray-700 dark:text-gray-300">
                  Remember me
                </Text>
              </label>
              <Link
                to="/forgot-password"
                className="text-sm text-primary hover:text-primary-600 dark:text-primary-400 dark:hover:text-primary-300 transition-colors duration-200"
              >
                Forgot password?
              </Link>
            </div>
            {/* Submit Button */}
            <Button
              type="submit"
              variant="primary"
              className="w-full h-12 text-lg font-semibold shadow-md rounded-lg transition-colors duration-200 bg-[#1A7F55] hover:bg-[#176945] focus:ring-[#1A7F55]/50 border-none"
              disabled={isLoading}
            >
              {isLoading ? 'Signing In...' : 'Sign In'}
            </Button>
          </form>
          {/* Arabic Verse */}
          <div className="text-center pt-4 border-t border-gray-200 dark:border-gray-700">
            <p lang="ar" dir="rtl" className="text-sm text-gray-600 dark:text-gray-400 mb-2">
              إِنَّمَا الْمُؤْمِنُونَ إِخْوَةٌ
            </p>
            <Text size="xs" muted className="text-center">
              "Indeed, the believers are brothers"
            </Text>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Login; 