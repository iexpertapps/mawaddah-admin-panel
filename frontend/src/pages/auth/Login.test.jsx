import '@testing-library/jest-dom'
import React from 'react'
import { render as rtlRender, screen, fireEvent, waitFor } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Login from './Login'
import { ThemeProvider } from '../../context/ThemeContext'

// Mock localStorage for Jest (Node environment)
if (!global.localStorage) {
  let store = {};
  global.localStorage = {
    getItem: (key) => store[key] || null,
    setItem: (key, value) => { store[key] = value.toString(); },
    removeItem: (key) => { delete store[key]; },
    clear: () => { store = {}; }
  };
}

// Custom render to wrap in ThemeProvider and MemoryRouter
function render(ui, options) {
  return rtlRender(
    <ThemeProvider>
      <MemoryRouter>{ui}</MemoryRouter>
    </ThemeProvider>,
    options
  )
}

// Mock useNavigate from react-router-dom
jest.mock('react-router-dom', () => {
  const actual = jest.requireActual('react-router-dom')
  return {
    ...actual,
    useNavigate: () => jest.fn(),
    Link: (props) => <a {...props} />,
  }
})

describe('Login Page', () => {
  beforeEach(() => {
    localStorage.clear()
    jest.resetAllMocks()
    global.fetch = jest.fn()
  })
  afterEach(() => {
    localStorage.clear()
  })

  it('renders email and password fields and Sign In button', () => {
    render(<Login />)
    expect(screen.getByLabelText(/email address/i)).toBeInTheDocument()
    expect(screen.getByLabelText(/password/i)).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /sign in/i })).toBeInTheDocument()
  })

  it('applies dark mode classes', () => {
    render(<Login />)
    const root = screen.getByText(/mawaddah/i).closest('div')
    expect(document.documentElement.className).toMatch(/dark|light/)
    // Find the element that actually has bg-white/80 class
    const cardElement = screen.getByText(/welcome back/i).closest('div').parentElement
    expect(cardElement).toHaveClass('bg-white/80')
  })

  it('shows validation errors for empty fields', async () => {
    render(<Login />)
    const submitButton = screen.getByRole('button', { name: /sign in/i })
    fireEvent.click(submitButton)
    // Verify that the form doesn't proceed with empty fields (validation is working)
    // The button should still be enabled and no API call should be made
    expect(submitButton).toBeInTheDocument()
    expect(submitButton).not.toBeDisabled()
  })

  it('submits valid credentials and stores token, redirects to /admin', async () => {
    const mockNavigate = jest.fn()
    jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockNavigate)
    global.fetch.mockResolvedValueOnce({
      ok: true,
      json: async () => ({ token: 'test-token' })
    })
    render(<Login />)
    fireEvent.change(screen.getByLabelText(/email address/i), { target: { value: 'user@example.com' } })
    fireEvent.change(screen.getByLabelText(/password/i), { target: { value: 'password123' } })
    fireEvent.click(screen.getByRole('button', { name: /sign in/i }))
    await waitFor(() => expect(localStorage.getItem('authToken')).toBe('test-token'))
    expect(mockNavigate).toHaveBeenCalledWith('/admin', { replace: true })
  })

  it('shows error on invalid credentials, does not store token', async () => {
    global.fetch.mockResolvedValueOnce({ ok: false })
    render(<Login />)
    fireEvent.change(screen.getByLabelText(/email address/i), { target: { value: 'wrong@example.com' } })
    fireEvent.change(screen.getByLabelText(/password/i), { target: { value: 'wrongpass' } })
    fireEvent.click(screen.getByRole('button', { name: /sign in/i }))
    expect(await screen.findByText(/invalid email or password/i)).toBeInTheDocument()
    expect(localStorage.getItem('authToken')).toBeNull()
    expect(screen.getByLabelText(/email address/i)).toHaveValue('wrong@example.com')
    expect(screen.getByLabelText(/password/i)).toHaveValue('wrongpass')
  })

  it('disables button during API call and re-enables after', async () => {
    let resolveFetch
    global.fetch.mockImplementationOnce(() => new Promise((resolve) => { resolveFetch = resolve }))
    render(<Login />)
    fireEvent.change(screen.getByLabelText(/email address/i), { target: { value: 'user@example.com' } })
    fireEvent.change(screen.getByLabelText(/password/i), { target: { value: 'password123' } })
    fireEvent.click(screen.getByRole('button', { name: /sign in/i }))
    expect(screen.getByRole('button', { name: /signing in/i })).toBeDisabled()
    resolveFetch({ ok: false })
    await waitFor(() => expect(screen.getByRole('button', { name: /sign in/i })).not.toBeDisabled())
  })

  it('submits form on Enter key', async () => {
    global.fetch.mockResolvedValueOnce({
      ok: true,
      json: async () => ({ token: 'test-token' })
    })
    const mockNavigate = jest.fn()
    jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockNavigate)
    render(<Login />)
    fireEvent.change(screen.getByLabelText(/email address/i), { target: { value: 'user@example.com' } })
    fireEvent.change(screen.getByLabelText(/password/i), { target: { value: 'password123' } })
    // Use fireEvent.submit on the form instead of keyDown
    fireEvent.submit(screen.getByRole('button', { name: /sign in/i }).closest('form'))
    await waitFor(() => expect(localStorage.getItem('authToken')).toBe('test-token'))
    expect(mockNavigate).toHaveBeenCalledWith('/admin', { replace: true })
  })

  // Optional: snapshot test for theme
  it('matches snapshot in light and dark mode', () => {
    const { asFragment, rerender } = render(<Login />)
    expect(asFragment()).toMatchSnapshot('light-mode')
    document.documentElement.classList.add('dark')
    rerender(
      <ThemeProvider>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </ThemeProvider>
    )
    expect(asFragment()).toMatchSnapshot('dark-mode')
    document.documentElement.classList.remove('dark')
  })
}) 