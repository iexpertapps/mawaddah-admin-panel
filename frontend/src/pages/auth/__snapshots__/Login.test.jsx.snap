// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Login Page matches snapshot in light and dark mode: dark-mode 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen flex items-center justify-center bg-gradient-to-br from-green-50 via-white to-accent-50 dark:from-gray-900 dark:via-gray-800 dark:to-gray-900 relative overflow-hidden"
  >
    <div
      class="absolute bottom-0 left-0 w-48 h-48 bg-maroon/20 blur-3xl rounded-full z-0 animate-fade-in"
    />
    <div
      class="absolute inset-0 opacity-5 dark:opacity-10 z-0"
    >
      <div
        class="absolute top-0 left-0 w-96 h-96 bg-primary rounded-full mix-blend-multiply filter blur-xl animate-pulse"
      />
      <div
        class="absolute top-0 right-0 w-96 h-96 bg-accent rounded-full mix-blend-multiply filter blur-xl animate-pulse animation-delay-2000"
      />
      <div
        class="absolute -bottom-8 left-20 w-72 h-72 bg-maroon rounded-full mix-blend-multiply filter blur-xl animate-pulse animation-delay-4000"
      />
    </div>
    <div
      class="absolute top-8 right-8 opacity-20 dark:opacity-30 animate-islamic-rotate z-0"
    >
      <svg
        class="text-primary"
        height="120"
        viewBox="0 0 120 120"
        width="120"
      >
        <defs>
          <pattern
            height="40"
            id="islamic-pattern"
            patternUnits="userSpaceOnUse"
            width="40"
            x="0"
            y="0"
          >
            <path
              d="M20 0L24 16L40 20L24 24L20 40L16 24L0 20L16 16Z"
              fill="currentColor"
              opacity="0.3"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#islamic-pattern)"
          height="120"
          width="120"
        />
      </svg>
    </div>
    <div
      class="relative z-10 w-full max-w-[400px] mx-auto px-6 py-8"
    >
      <div
        class="bg-white/80 dark:bg-gray-800/80 backdrop-blur-sm rounded-2xl shadow-2xl border border-white/20 dark:border-gray-700/20 px-6 py-8 space-y-8"
      >
        <div
          class="text-center space-y-4"
        >
          <div
            class="flex justify-center"
          >
            <div
              class="relative animate-float transition-all duration-500 hover:scale-105 shadow-[0_0_12px_rgba(26,127,85,0.6)]"
            >
              <img
                alt="Mawaddah Logo"
                class="h-16 w-16 rounded-xl object-contain mx-auto select-none"
                draggable="false"
                src="/ic_mawaddah_180x180.png"
              />
              <div
                class="absolute -inset-1 bg-gradient-to-r from-accent to-primary rounded-2xl opacity-20 blur-sm pointer-events-none"
              />
            </div>
          </div>
          <div
            class="space-y-2"
          >
            <h1
              class="text-2xl font-bold text-gray-900 dark:text-gray-100 text-center bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent"
            >
              Mawaddah
            </h1>
            <p
              class="text-sm text-gray-500 dark:text-gray-400 text-center"
            >
              Admin Dashboard
            </p>
          </div>
        </div>
        <div
          class="text-center space-y-2"
        >
          <h2
            class="text-lg font-semibold text-gray-900 dark:text-gray-100 text-gray-900 dark:text-gray-100"
          >
            Welcome Back
          </h2>
          <p
            class="text-base text-gray-500 dark:text-gray-400 text-sm"
          >
            Sign in to your account to continue
          </p>
        </div>
        <form
          class="space-y-6"
        >
          <div
            class="space-y-4"
          >
            <div
              class="w-full"
            >
              <label
                class="block mb-1 text-sm font-medium text-gray-700 dark:text-gray-300"
                for="email"
              >
                Email Address
              </label>
              <input
                class="
          w-full rounded-lg border bg-white dark:bg-gray-900 text-gray-900 dark:text-gray-100 placeholder-gray-400 dark:placeholder-gray-500 focus:ring-2 focus:ring-primary focus:border-primary transition-all duration-200
          px-4 py-2
          border-gray-300 dark:border-gray-700
          
        "
                id="email"
                placeholder="Enter your email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="w-full"
            >
              <label
                class="block mb-1 text-sm font-medium text-gray-700 dark:text-gray-300"
                for="password"
              >
                Password
              </label>
              <input
                class="
          w-full rounded-lg border bg-white dark:bg-gray-900 text-gray-900 dark:text-gray-100 placeholder-gray-400 dark:placeholder-gray-500 focus:ring-2 focus:ring-primary focus:border-primary transition-all duration-200
          px-4 py-2
          border-gray-300 dark:border-gray-700
          
        "
                id="password"
                placeholder="Enter your password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <label
              class="flex items-center space-x-2 cursor-pointer"
            >
              <input
                class="w-4 h-4 text-primary border-gray-300 rounded focus:ring-primary focus:ring-2"
                type="checkbox"
              />
              <p
                class="text-sm text-gray-700 dark:text-gray-300 text-gray-700 dark:text-gray-300"
              >
                Remember me
              </p>
            </label>
            <a
              class="text-sm text-primary hover:text-primary-600 dark:text-primary-400 dark:hover:text-primary-300 transition-colors duration-200"
              to="/forgot-password"
            >
              Forgot password?
            </a>
          </div>
          <button
            class="font-medium rounded-lg transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 bg-green-600 hover:bg-green-700 text-white focus:ring-green-500/50 dark:bg-green-600 dark:hover:bg-green-700 px-6 py-3 text-lg w-full hover:scale-105 transition-transform duration-200 hover:shadow-inner"
            type="submit"
          >
            Sign In
          </button>
        </form>
        <div
          class="text-center pt-4 border-t border-gray-200 dark:border-gray-700"
        >
          <p
            class="text-center text-sm mt-6 italic text-gray-400"
            dir="rtl"
            lang="ar"
          >
            إِنَّمَا الْمُؤْمِنُونَ إِخْوَةٌ
          </p>
          <p
            class="text-xs text-gray-500 dark:text-gray-400 mt-1"
          >
            "Indeed, the believers are brothers"
          </p>
        </div>
      </div>
      <div
        class="absolute -bottom-4 -left-4 opacity-20 dark:opacity-30 z-0"
      >
        <svg
          class="text-accent"
          height="80"
          viewBox="0 0 80 80"
          width="80"
        >
          <defs>
            <pattern
              height="20"
              id="bottom-pattern"
              patternUnits="userSpaceOnUse"
              width="20"
              x="0"
              y="0"
            >
              <circle
                cx="10"
                cy="10"
                fill="currentColor"
                opacity="0.4"
                r="2"
              />
            </pattern>
          </defs>
          <rect
            fill="url(#bottom-pattern)"
            height="80"
            width="80"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login Page matches snapshot in light and dark mode: light-mode 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen flex items-center justify-center bg-gradient-to-br from-green-50 via-white to-accent-50 dark:from-gray-900 dark:via-gray-800 dark:to-gray-900 relative overflow-hidden"
  >
    <div
      class="absolute bottom-0 left-0 w-48 h-48 bg-maroon/20 blur-3xl rounded-full z-0 animate-fade-in"
    />
    <div
      class="absolute inset-0 opacity-5 dark:opacity-10 z-0"
    >
      <div
        class="absolute top-0 left-0 w-96 h-96 bg-primary rounded-full mix-blend-multiply filter blur-xl animate-pulse"
      />
      <div
        class="absolute top-0 right-0 w-96 h-96 bg-accent rounded-full mix-blend-multiply filter blur-xl animate-pulse animation-delay-2000"
      />
      <div
        class="absolute -bottom-8 left-20 w-72 h-72 bg-maroon rounded-full mix-blend-multiply filter blur-xl animate-pulse animation-delay-4000"
      />
    </div>
    <div
      class="absolute top-8 right-8 opacity-20 dark:opacity-30 animate-islamic-rotate z-0"
    >
      <svg
        class="text-primary"
        height="120"
        viewBox="0 0 120 120"
        width="120"
      >
        <defs>
          <pattern
            height="40"
            id="islamic-pattern"
            patternUnits="userSpaceOnUse"
            width="40"
            x="0"
            y="0"
          >
            <path
              d="M20 0L24 16L40 20L24 24L20 40L16 24L0 20L16 16Z"
              fill="currentColor"
              opacity="0.3"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#islamic-pattern)"
          height="120"
          width="120"
        />
      </svg>
    </div>
    <div
      class="relative z-10 w-full max-w-[400px] mx-auto px-6 py-8"
    >
      <div
        class="bg-white/80 dark:bg-gray-800/80 backdrop-blur-sm rounded-2xl shadow-2xl border border-white/20 dark:border-gray-700/20 px-6 py-8 space-y-8"
      >
        <div
          class="text-center space-y-4"
        >
          <div
            class="flex justify-center"
          >
            <div
              class="relative animate-float transition-all duration-500 hover:scale-105 shadow-[0_0_12px_rgba(26,127,85,0.6)]"
            >
              <img
                alt="Mawaddah Logo"
                class="h-16 w-16 rounded-xl object-contain mx-auto select-none"
                draggable="false"
                src="/ic_mawaddah_180x180.png"
              />
              <div
                class="absolute -inset-1 bg-gradient-to-r from-accent to-primary rounded-2xl opacity-20 blur-sm pointer-events-none"
              />
            </div>
          </div>
          <div
            class="space-y-2"
          >
            <h1
              class="text-2xl font-bold text-gray-900 dark:text-gray-100 text-center bg-gradient-to-r from-primary to-accent bg-clip-text text-transparent"
            >
              Mawaddah
            </h1>
            <p
              class="text-sm text-gray-500 dark:text-gray-400 text-center"
            >
              Admin Dashboard
            </p>
          </div>
        </div>
        <div
          class="text-center space-y-2"
        >
          <h2
            class="text-lg font-semibold text-gray-900 dark:text-gray-100 text-gray-900 dark:text-gray-100"
          >
            Welcome Back
          </h2>
          <p
            class="text-base text-gray-500 dark:text-gray-400 text-sm"
          >
            Sign in to your account to continue
          </p>
        </div>
        <form
          class="space-y-6"
        >
          <div
            class="space-y-4"
          >
            <div
              class="w-full"
            >
              <label
                class="block mb-1 text-sm font-medium text-gray-700 dark:text-gray-300"
                for="email"
              >
                Email Address
              </label>
              <input
                class="
          w-full rounded-lg border bg-white dark:bg-gray-900 text-gray-900 dark:text-gray-100 placeholder-gray-400 dark:placeholder-gray-500 focus:ring-2 focus:ring-primary focus:border-primary transition-all duration-200
          px-4 py-2
          border-gray-300 dark:border-gray-700
          
        "
                id="email"
                placeholder="Enter your email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="w-full"
            >
              <label
                class="block mb-1 text-sm font-medium text-gray-700 dark:text-gray-300"
                for="password"
              >
                Password
              </label>
              <input
                class="
          w-full rounded-lg border bg-white dark:bg-gray-900 text-gray-900 dark:text-gray-100 placeholder-gray-400 dark:placeholder-gray-500 focus:ring-2 focus:ring-primary focus:border-primary transition-all duration-200
          px-4 py-2
          border-gray-300 dark:border-gray-700
          
        "
                id="password"
                placeholder="Enter your password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="flex items-center justify-between"
          >
            <label
              class="flex items-center space-x-2 cursor-pointer"
            >
              <input
                class="w-4 h-4 text-primary border-gray-300 rounded focus:ring-primary focus:ring-2"
                type="checkbox"
              />
              <p
                class="text-sm text-gray-700 dark:text-gray-300 text-gray-700 dark:text-gray-300"
              >
                Remember me
              </p>
            </label>
            <a
              class="text-sm text-primary hover:text-primary-600 dark:text-primary-400 dark:hover:text-primary-300 transition-colors duration-200"
              to="/forgot-password"
            >
              Forgot password?
            </a>
          </div>
          <button
            class="font-medium rounded-lg transition-colors focus:outline-none focus:ring-2 focus:ring-offset-2 bg-green-600 hover:bg-green-700 text-white focus:ring-green-500/50 dark:bg-green-600 dark:hover:bg-green-700 px-6 py-3 text-lg w-full hover:scale-105 transition-transform duration-200 hover:shadow-inner"
            type="submit"
          >
            Sign In
          </button>
        </form>
        <div
          class="text-center pt-4 border-t border-gray-200 dark:border-gray-700"
        >
          <p
            class="text-center text-sm mt-6 italic text-gray-400"
            dir="rtl"
            lang="ar"
          >
            إِنَّمَا الْمُؤْمِنُونَ إِخْوَةٌ
          </p>
          <p
            class="text-xs text-gray-500 dark:text-gray-400 mt-1"
          >
            "Indeed, the believers are brothers"
          </p>
        </div>
      </div>
      <div
        class="absolute -bottom-4 -left-4 opacity-20 dark:opacity-30 z-0"
      >
        <svg
          class="text-accent"
          height="80"
          viewBox="0 0 80 80"
          width="80"
        >
          <defs>
            <pattern
              height="20"
              id="bottom-pattern"
              patternUnits="userSpaceOnUse"
              width="20"
              x="0"
              y="0"
            >
              <circle
                cx="10"
                cy="10"
                fill="currentColor"
                opacity="0.4"
                r="2"
              />
            </pattern>
          </defs>
          <rect
            fill="url(#bottom-pattern)"
            height="80"
            width="80"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
